ramips: Add support command fw_setsys for Xiaomi routers

remittor opened the pull request:
The system parameters are contained in the `Bdata` partition.
To use the `fw_setsys` command, you need to create a file `fw_sys.config`. This file is created after calling the functions `ubootenv_add_uci_sys_config` and `ubootenv_add_app_config`.

adschm commented:
This has formal issues. Please use your real name for Signed-off-by and wrap you commit message so it has 74 chars. max per line.

remittor commented on package/boot/uboot-envtools/files/ramips:
```diff
@@ -73,6 +76,6 @@ zyxel,nr7101)
 
 config_load ubootenv
-config_foreach ubootenv_add_app_config ubootenv
+config_foreach ubootenv_add_app_config
```
It's not a wrong.
Look this: https://github.com/openwrt/openwrt/commit/a3e9fd7e5b365aa2f0b8cc1dc0a2d7cf3daa3e1a#diff-3b87ce92c70f9b47215c6c7e7e0afbbe39c95e62b815935e95220dd12b124ca4

remittor commented:
> This has formal issues.

Fixed

DragonBluep commented on package/boot/uboot-envtools/files/ramips:
```diff
@@ -57,13 +54,19 @@ linksys,ea7300-v1|\
 xiaomi,redmi-router-ac2100)
-	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x1000" "0x20000"
+	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x1000" "0x10000"
+	ubootenv_add_uci_sys_config "/dev/mtd2" "0x0" "0x4000" "0x10000"
```
Maybe secsize for nand devices needs to be set to 0x20000 (128KiB) ?

Ansuel replied:
@remittor why from
```
ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x1000" "0x20000"
```

to 

```
ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x1000" "0x10000"
```

remittor replied:
Because I don't want to break something on device `xiaomi,mi-router-4a-gigabit` (64 KiB sector)

remittor replied:
> Can't we just extend the switch and apply the correct sector based on the device to be extra safe?

OK. Let's do that. Now I'll try to fix it.

Ansuel commented:
Also don't know if `Oleg S` is acceptable. Is it really a problem here?

Ansuel commented:
@remittor Still interested in this?

remittor commented:
> @remittor Still interested in this?

Yes

remittor committed:
ramips: Add support command fw_setsys for Xiaomi routers

The system parameters are contained in the Bdata partition.
To use the fw_setsys command, you need to create a file fw_sys.config. This file is created after calling the functions ubootenv_add_uci_sys_config and ubootenv_add_app_config.

Signed-off-by: Oleg S <remittor@gmail.com>

Ansuel commented:
Thanks, merged in master with minor changes 6c7e337c80f92693c2ca628a4a56aeaec4cc3ca8
